typescript/sdk/src/ism/metadata/arbL2ToL1.ts

```diff
@@ -0,0 +1,169 @@
+  data: BytesLike;
+}
+
+const ArbSys = ArbSys__factory.createInterface();
+
+export class ArbL2ToL1MetadataBuilder implements MetadataBuilder {
+  constructor(
+    protected readonly core: HyperlaneCore,
+    protected readonly logger = rootLogger.child({
+      module: 'ArbL2ToL1MetadataBuilder',
+    }),
+  ) {}
+
+  async build(
+    context: MetadataContext<
+      WithAddress<ArbL2ToL1IsmConfig>,
+      WithAddress<ArbL2ToL1HookConfig>
+    >,
+  ): Promise<string> {
+    assert(context.ism.type === IsmType.ARB_L2_TO_L1, 'Invalid ISM type');
+    this.logger.debug({ context }, 'Building ArbL2ToL1 metadata');
+
+    // if the executeTransaction call is already successful, we can call with null metadata
+    const ism = AbstractMessageIdAuthorizedIsm__factory.connect(
+      context.ism.address,
+      this.core.multiProvider.getSigner(context.message.parsed.destination),
+    );
+    const verified = await ism.isVerified(context.message.id);
+    if (verified) {
+      this.logger.debug(
+        'Message is already verified, relaying without metadata...',
+      );
+      return '0x';
+    }
+
+    // else build the metadata for outbox.executeTransaction call
+    const matchingL2Tx = context.dispatchTx.logs
+      .filter((log) => eqAddressEvm(log.address, context.hook.arbSys))
+      .map((log) => ArbSys.parseLog(log))
+      .find((log) => {
+        const calldata: string = log.args.data;
+        const messageIdHex = context.message.id.slice(2);
+        return calldata && calldata.includes(messageIdHex);
+      });
+
+    assert(matchingL2Tx, 'No matching L2ToL1Tx event found');
+    this.logger.debug({ matchingL2Tx }, 'Found matching L2ToL1Tx event');
+
+    if (matchingL2Tx) {
+      const l2ToL1TxEvent: L2ToL1TransactionEvent = {
+        ...matchingL2Tx.args,
+        caller: matchingL2Tx.args.caller,
+        destination: matchingL2Tx.args.destination,
+        hash: matchingL2Tx.args.hash,
+        position: matchingL2Tx.args.position,
+        arbBlockNum: matchingL2Tx.args.arbBlockNum,
+        ethBlockNum: matchingL2Tx.args.ethBlockNum,
+        timestamp: matchingL2Tx.args.timestamp,
+        callvalue: matchingL2Tx.args.callvalue,
+        data: matchingL2Tx.args.data,
+      };
+
+      const reader = new L2ToL1MessageReader(
+        this.core.multiProvider.getProvider('sepolia'),
+        l2ToL1TxEvent,
+      );
+
+      const baseProvider = new providers.JsonRpcProvider(
+        this.core.multiProvider.getChainMetadata(
+          'arbitrumsepolia',
+        ).rpcUrls[0].http,
+      );
+      const arbProvider = new ArbitrumProvider(baseProvider, {
+        name: 'arbitrum-sepolia',
+        chainId: 421614,
+      });
+
+      const status = await reader.status(arbProvider);
+      // need to wait for the challenge period to pass before relaying
+      if (!status) {
+        throw new Error(
+          `Arbitrum L2ToL1 message isn't ready for relay. Wait until the challenge period before relaying again.`,
```

Comment:
any way we can surface timestamp information here?

Reply:
It's a static value, always showing ~90 blocks.

Reply:
the function returns that yes
to be more explicit, my suggestion is to calculate the remaining time by checking the origin timestamp, adding the fraud window duration, and printing that in human readable format

Reply:
I dont think number of blocks will be very readable for the user but maybe this is good enough for now